### PR TITLE
feat: update block filters UI in materials

### PR DESCRIPTION
**Pull request by gene9831**
English | [简体中文](https://github.com/opentiny/tiny-engine/blob/develop/.github/PULL_REQUEST_TEMPLATE/PULL_REQUEST_TEMPLATE.zh-CN.md)

# PR

## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] The commit message follows our [Commit Message Guidelines](https://github.com/opentiny/tiny-engine/blob/develop/CONTRIBUTING.md)
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)
- [ ] Built its own designer, fully self-validated

## PR Type

What kind of change does this PR introduce?

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] Other... Please describe:

## Background and solution
在物料插件里面的添加区块页面中，原来的区块筛选器是checkbox group的形式。如果checkbox数量很多，则会显示得很凌乱。现优化成select组件

### What is the current behavior?

image

### What is the new behavior?

image

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

## Summary by CodeRabbit

- **New Features**
	- Introduced a dynamic filtering mechanism with checkboxes and selection options for enhanced user interaction.
	- Added a `usingSelect` property to filters, enabling a selection interface for specific filters.

- **Bug Fixes**
	- Improved error handling for the block addition process, ensuring users are informed of any issues.

- **Style**
	- Enhanced layout and spacing for filter components.
	- Adjusted the footer toolbar for better alignment.
	- Updated styling for tags, dropdowns, and input components to improve visual consistency and user experience.
	- Refined hover and focus states for better user interaction feedback.

### CODE REVIEW

File: packages/plugins/materials/src/meta/block/src/BlockGroupFilters.vue [hunk 1]
```diff
@@ -4,25 +4,42 @@
...
+          v-model="state.checkGroup[filter.id]"
+          size="mini"
+          multiple
+          @change="getFilters($event, filter.id)"
+        >
+          <tiny-option
+            v-for="item in selectOptions[filter.id]"
+            :key="item.name"
+            :label="item.name"
+            :value="item"
```
Comment (hexqi) [inline packages/plugins/materials/src/meta/block/src/BlockGroupFilters.vue]: 这里应该绑定到`item.value`?

Comment (gene9831) [reply packages/plugins/materials/src/meta/block/src/BlockGroupFilters.vue]: 已修改

**Review by chilingling — DISMISSED** [review]

File: packages/plugins/materials/src/meta/block/src/BlockGroupFilters.vue [hunk 2]
```diff
@@ -94,6 +134,12 @@ export default {
           display: none;
         }
       }
+      :deep(.tiny-select.tiny-select .tiny-select__tags) {
+        max-width: calc(100% - 24px) !important;
+        .tiny-tag {
+          background-color: var(--te-common-bg-container);
```
Comment (wenmine) [inline packages/plugins/materials/src/meta/block/src/BlockGroupFilters.vue]: 这个需要跟别歌确认一下，用的是container还是--te-common-bg-disabled

Comment (gene9831) [reply packages/plugins/materials/src/meta/block/src/BlockGroupFilters.vue]: 已确认，使用`--te-common-bg-disabled`

**Review by chilingling — APPROVED** [review]

**Review by hexqi — APPROVED** [review]

**Pull request merged**